File: WindowsForm_sample/UITest/FormSession.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium.Appium.Windows;
using OpenQA.Selenium.Remote;
using OpenQA.Selenium;
using Microsoft.ServiceBus.Configuration;
using System;
using System.IO;

namespace UITestSample.Test
{
    // 初期設定、最後の後処理をするクラス  
    // シナリオテストの継承元のクラス  
    public class FormSession
    {
        protected const string WindowsApplicationDriverUrl = "http://127.0.0.1:4723";
        private static readonly string NotepadAppId = Path.Combine(Environment.CurrentDirectory, "WindowsForm_sample.exe");
        protected static WindowsDriver<OpenQA.Selenium.Appium.Windows.WindowsElement> session;
        protected static OpenQA.Selenium.Appium.Windows.WindowsElement editBox;

        public static void Setup(TestContext context)
        {
            if (session == null)
            {
                DesiredCapabilities appCapabilities = new OpenQA.Selenium.Remote.DesiredCapabilities();
                appCapabilities.SetCapability("app", NotepadAppId);
                session = new WindowsDriver<OpenQA.Selenium.Appium.Windows.WindowsElement>(new Uri(WindowsApplicationDriverUrl), appCapabilities);
                Assert.AreEqual("Form1", session.Title);
            }
        }

        // セッションを切って　アプリケーションを閉じる  
        public static void TearDown()
        {
            // Close the application and delete the session  
            if (session != null)
            {
                session.Close();
                session.Quit();
                session = null;
            }
        }
    }
}

File: VBMSTest/FormSession.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports OpenQA.Selenium.Appium.Windows
Imports OpenQA.Selenium.Remote
Imports System
Imports System.IO

Namespace VBMSTest
    ' 初期設定と後処理を行う基底クラス
    Public Class FormSession
        Protected Const WindowsApplicationDriverUrl As String = "http://127.0.0.1:4723"

        ' テストのプロジェクト名を取得
        Private Shared ReadOnly testProjectName As String = System.Reflection.Assembly.GetExecutingAssembly().GetName().Name
        Private Shared ReadOnly solutionPath As String = AppContext.BaseDirectory.Substring(0, AppContext.BaseDirectory.IndexOf(testProjectName))
        Private Shared ReadOnly NotepadAppId As String = Path.Combine(solutionPath + "\VBWinform\bin\Debug", "VBWinform.exe")
        Protected Shared session As WindowsDriver(Of WindowsElement)
        Protected Shared editBox As WindowsElement

        Public Shared Sub Setup()
            If session Is Nothing Then
                Dim appCapabilities As New DesiredCapabilities()
                appCapabilities.SetCapability("app", NotepadAppId)
                session = New WindowsDriver(Of WindowsElement)(New Uri(WindowsApplicationDriverUrl), appCapabilities)
                Assert.AreEqual("Form1", session.Title)
            End If
        End Sub

        ' セッションを終了し、アプリケーションを閉じる
        Public Shared Sub TearDown()
            If session IsNot Nothing Then
                session.Close()
                session.Quit()
                session = Nothing
            End If
        End Sub
    End Class
End Namespace
